### DISCUSSION TITLE
Database auto-locking

### DISCUSSION THREAD

**Discussion by Chaython** [keepassxreboot/keepassxc#7281] [Q&A] [answered]
Is the auto locking, security feature, to protect against, physical intrusion [user afk] or does it also protect against in memory attacks?
It's just annoying to have to keep unlocking the db.

**Comment by droidmonkey** [comment, marked as answer]
It does both. When your database is locked, all memory used for the previously loaded database is securely wiped.

**Reply by Chaython** [reply]
So for convivence/security, would it better to have an offline DB, with an easier password or have my normal DB/password, with a longer lockout time?

**Reply by droidmonkey** [reply]
That is entirely up to your risk appetite. Higher stakes credentials would ideally be in a separate database with more complex credentials. We don't have per-database lockout settings yet.
